rfcs/0000-kuadrant-ci-cd.md

```diff
@@ -0,0 +1,832 @@
+cp [path-to]/kuadrant-operator/bundle.Dockerfile [path-to]/community-operators/operators/kuadrant-operator/0.5.0/
+```
+
+3. Create a PR to the `community-operators` (same for `community-operators-prod`) repository and follow the guidelines
+exposed in the auto generated description of the PR.
+
+</details>
+
+## New process
+The new process ideally should be as simple as possible, and should be able to be triggered by any person, process and
+set of events (particular commits, tags, merge, etc) that needs the artifacts. It also should be able to be repeated
+in a consistent way and without the need of any human setting variables or parameters at the time of triggering it
+(at least for the most common cases).
+
+Given our current process, we can identify some similarities and particularities between the different components:
+
+* All of them are built using a CI/CD tool (GitHub Actions) and published to a container registry (quay.io).
+* Not all of them are built using the same build system (Cargo, Go, Operator SDK, etc.).
+* 3 of them follow the Operator Pattern and are built using the Operator SDK (Authorino, Limitador and Kuadrant operators).
+* 2 of them are Rust projects (Limitador and WASMShim).
+* There's a dependency between the different components, Kuadrant Operator depends on Authorino and Limitador Operators,
+and the WasmShim; while the before mentioned operators depend on their operands (Authorino and Limitador).
+* There's no clear steps defined regarding the approval of the deliverables (images, manifests, etc.) and their
+publication to our website, Github releases, OperatorHub, etc.
+
+Given the before mentioned points, we can identify some common steps that can be extracted and reused for all the
+components, and some particularities that need to be addressed in a different way, but the overall
+process should be the same for all of them in terms of triggering it and the steps that need to be followed.
+
+### Option 1: Release File
+The first option is to create a file in the root of each repository that contains the information needed to build and
+release the component. This file could be a YAML, TOML or simply a key-value file. The file should contain the following
+information:
+
+* Component/s name/s and version/s (In the case of Limitador, it should contain the name and version of the crate and the name of the server).
+* Dependencies name/s and version/s.
+* Replace version (in the case of the Operators, this version is needed for the OLM upgrade process).
+
+This file should be updated every time a new release is made, and it should be committed to the repository to the release
+branch. This file will be used by the CI/CD tool to build and release the component. In the case of the main branch, the
+file will include the next version to be released, following the floating suffix `-dev` for its version and its dependencies.
```

Comment:
I like the `-dev` (or some equivalent suffix) for "in progress" work... another option is to use the git SHA as [a suffix](https://www.reddit.com/r/devops/comments/ax84t0/comment/ehs2rrz/) - tho I'd prefix it with something stable so semantic versioning (as pointed in the comments of that thread). Or put the sha in the [metadata](https://semver.org/#spec-item-10), as:

>Build metadata MUST be ignored when determining version precedence.

That's if we wanted to move keep some sort of "floating release" while achieving proper tracability

So using -dev, also would prevent us from using `-alpha` or `-beta` ... or anything "lower than `-dev`, as per [semver](https://semver.org/#spec-item-11)

>Identifiers with letters or hyphens are compared lexically in ASCII sort order.